File: UpravljanjeSZ/test/javascript/pages/zgrade/ZgradaPregledPage.js
//konstruktor stranice
var ZgradaPregledPage = function () { }
var utils = require('../utils.js');
// Opis polja i metoda
ZgradaPregledPage.prototype = Object.create({}, {

     // dobavljanje dodavanje dugmeta
     btnDodavanje: {
        get: function() {
            return utils.waitForElementPresence(by.xpath('//button[@class="btn btn-outline-primary"]'), 10000);
        },
    },
    // dobavljanje pregled dugmeta
    btnPregled: {
        get: function() {
            return utils.waitForElementPresence(by.xpath('//button[@class="btn btn-primary"]'),10000);
        },
    },
    //SELEKT
    prikaz: {
        get: function() {
            return utils.waitForElementPresence(by.id('prikaz'), 10000).getText();
        },
        set: function(value) {
            this.prikaz.element(by.cssContainingText('option', value)).click();
        }
    },
    // PRETRAGA
	// dobavljanje i setovanje polja ulica ili broj
    ulicaBroj: {
        get: function() {
            return utils.waitForElementPresence(by.id('ulicaBroj'), 10000);
        },
        set: function(value) {
            return this.ulicaBroj.clear().sendKeys(value);
        }
    },
    // dobavljanje i setovanje grad polja
    grad: {
        get: function() {
            return utils.waitForElementPresence(by.id('mesto'), 10000);
        },
        set: function(value) {
            return this.grad.clear().sendKeys(value);
        }
    },
    // dobavljanje pretraga dugmeta
    btnPretraga: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("//div[@class='row']//button"),10000);
        },
    },
    // dobavljanje poruke pretrage
    getMsgPretraga: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("/html/body/app-root/app-zgrade/div/app-izlistaj-zgrade/div/div[1]/h2"),10000);
        },
    },
    //metoda za pretragu za oba polja unesena
    pretraga: {
        value: function(ulicabroj, grad) {
            this.ulicaBroj = ulicabroj;
            this.grad = grad;
            this.btnPretraga.click();
        }
    },

    // TABELA
    // cela tabela zgrade
    zgradaTable: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("/html/body/app-root/app-zgrade/div/app-izlistaj-zgrade/div/div[1]/table"), 10000);
        }
    },
    // red moje zgrade izvucen iz tabele po adresi
    getZgraduIzTabele: {
        value: function(adresa) {
            return utils.waitForElementPresence(by.xpath('//*[contains(text(),"' + adresa + '")]/../..'),10000);
        }
    },

    //adresa link u tabeli
    linkAdresa: {
        get: function() {
            return utils.waitForElementPresence(by.xpath('/html/body/app-root/app-zgrade/div/app-izlistaj-zgrade/div/div[1]/table/tbody/tr[1]/td[1]/a'),10000);
        },
    },
    //predsednik link u tabeli
    linkPredsednik: {
        get: function() {
            return utils.waitForElementPresence(by.xpath('/html/body/app-root/app-zgrade/div/app-izlistaj-zgrade/div/div[1]/table/tbody/tr[1]/td[2]/a'),10000);
        },
    },
});

// Export klase
module.exports = ZgradaPregledPage;

File: UpravljanjeSZ/main/java/pages/zgrade/ZgradaPregledPage.java
package pages.zgrade;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;

import pages.helpers.Utility;

public class ZgradaPregledPage {
	private WebDriver driver;

	// konstruktor stranice
	public ZgradaPregledPage(WebDriver driver) {
		this.driver = driver;
	}

	// dobavljanje dodavanje dugmeta
	public WebElement getBtnDodavanje() {
		return Utility.waitForElementPresence(driver, By.xpath("//button[@class='btn btn-outline-primary']"), 10);
	}

	// dobavljanje pregled dugmeta
	public WebElement getBtnPregled() {
		return Utility.waitForElementPresence(driver, By.xpath("//button[@class='btn btn-primary']"), 10);
	}
	//SELEKT
	public Select getPrikaz() {
		return new Select(Utility.waitForElementPresence(driver, By.id("prikaz"), 10));
	}
	public void setPrikaz(String value) {
		this.getPrikaz().selectByVisibleText(value);
	}
    
	// PRETRAGA
	// dobavljanje i setovanje polja ulica ili broj
	public WebElement getUlicaBroj() {
		return Utility.waitForElementPresence(driver, By.id("ulicaBroj"), 10);
	}

	public void setUlicaBroj(String ulicabroj) {
		WebElement ulicaBroj = getUlicaBroj();
		ulicaBroj.clear();
		ulicaBroj.sendKeys(ulicabroj);
	}

	// dobavljanje i setovanje grad polja
	public WebElement getGrad() {
		return Utility.waitForElementPresence(driver, By.id("mesto"), 10);
	}

	public void setGrad(String mesto) {
		WebElement grad = getGrad();
		grad.clear();
		grad.sendKeys(mesto);
	}

	// dobavljanje pretraga dugmeta
	public WebElement getBtnPretraga() {
		return Utility.waitForElementPresence(driver, By.xpath("//div[@class='row']//button"), 10);
	}
	//metoda za pretragu za oba polja unesena
	public void pretraga(String ulicaBroj, String grad){
		setUlicaBroj(ulicaBroj);
		setGrad(grad);
		getBtnPretraga().click();
	}
	// dobavljanje poruke pretrage
		public WebElement getMsgPretraga() {
			return Utility.waitForElementPresence(driver, By.xpath("/html/body/app-root/app-zgrade/div/app-izlistaj-zgrade/div/div[1]/h2"), 10);
		}
	
	// TABELA
	// cela tabela zgrade
	public WebElement getZgradeTable() {
		return Utility.waitForElementPresence(driver, By.xpath("//table[@class='table table-hover ng-star-inserted']"), 10);
	}

	// red moje zgrade izvucen iz tabele po nazivu
	public WebElement getZgraduIzTabele(String adresa) {
		return Utility.waitForElementPresence(driver, By.xpath("//*[contains(text(),\"" + adresa + "\")]/../.."), 10);
	}
	//metoda za proveru da li zgrada postoji u tabeli
	public boolean isZgradaPresent(String adresa){
		return Utility.isPresent(driver, By.xpath("//*[contains(text(),\"" + adresa + "\")]/../.."));
	}
	//adresa link u tabeli
	public WebElement getLinkAdresa() {
		return Utility.waitForElementPresence(driver, By.xpath("/html/body/app-root/app-zgrade/div/app-izlistaj-zgrade/div/div[1]/table/tbody/tr[1]/td[1]/a"), 10);
	}
	//predsednik link u tabeli
	public WebElement getLinkPredsednik() {
		return Utility.waitForElementPresence(driver, By.xpath("/html/body/app-root/app-zgrade/div/app-izlistaj-zgrade/div/div[1]/table/tbody/tr[1]/td[2]/a"), 10);
	}
}
